azurelinuxagent/pa/rdma/factory.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2016 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.4+ and Openssl 1.0+
+#
+
+import azurelinuxagent.common.logger as logger
+
+from azurelinuxagent.common.version import DISTRO_NAME
+from azurelinuxagent.common.rdma import RDMAHandler
+from .suse import SUSERDMAHandler
+
+
+def get_rdma_handler(distro_name=DISTRO_NAME, distro_version=DISTRO_VERSION):
```

Comment:
everything which is under common/osutil has already been through the appropriate factory. let's keep a single set of these checks, and move this code under there. I would also suggest we move rdma.py to common/rdma.py rather than pa.